### PR TITLE
Use more `cppwinrt_utils`

### LINKED ISSUE #2456: terminalapp, terminalconnection: use `cppwinrt_utils.h` macros where possible
# Description of the new feature/enhancement

Make sure we're using the `cppwinrt_utils.h` macros everywhere we can. And if we have a lot of repeated code in some places, feel free to introduce new macros.

### PR DESCRIPTION
**Pull request by zadjii-msft**
* Adds the `cppwinrt_utils.h` to the include path for all winrt projects.
* Adds the file to the pch.h's for every project, so you don't need to include it in every header.
* Replaces some usages of `DECLARE_EVENT`/`DEFINE_EVENT` with `WINRT_CALLBACK`, which will do it in a oneliner
* Adds more `BASIC_FACTORY` usage.
* [x] Closes #2456

It's 92 files with one-line changes, don't be scared.

### CODE REVIEW

File: src/cppwinrt.build.pre.props [hunk 1]
```diff
@@ -78,6 +78,8 @@
       <DisableSpecificWarnings>5104;28204;%(DisableSpecificWarnings)</DisableSpecificWarnings>
 
       <AdditionalUsingDirectories>$(WindowsSDK_WindowsMetadata);$(AdditionalUsingDirectories)</AdditionalUsingDirectories>
+
+      <AdditionalIncludeDirectories>$(OpenConsoleDir)src\cascadia\inc;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (lhecker) [inline src/cppwinrt.build.pre.props:82]: There's already a `<AdditionalIncludeDirectories>` below where this could go.

**Review by lhecker — APPROVED** [review]

File: src/cascadia/TerminalApp/TabRowControl.h [hunk 2]
```diff
@@ -25,7 +24,5 @@ namespace winrt::TerminalApp::implementation
 
 namespace winrt::TerminalApp::factory_implementation
 {
-    struct TabRowControl : TabRowControlT<TabRowControl, implementation::TabRowControl>
-    {
-    };
+    BASIC_FACTORY(TabRowControl);
```
Comment (miniksa) [inline src/cascadia/TerminalApp/TabRowControl.h:27]: Outside the scope of this... but maybe we should prefix these macros with something (like how the WIL ones do WI_FLAG_SET) so we understand which ones are a part of our magic and which ones are a part of the actual cppwinrt packages.

File: src/cppwinrt.build.pre.props [hunk 3]
```diff
@@ -87,7 +87,7 @@
 
   <ItemDefinitionGroup>
     <ClCompile>
-      <AdditionalIncludeDirectories>$(OpenConsoleDir)\src\cascadia\WinRTUtils\inc;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(OpenConsoleDir)\src\cascadia\WinRTUtils\inc;$(OpenConsoleDir)src\cascadia\inc;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (miniksa) [inline src/cppwinrt.build.pre.props:90]: I mean, at this point... maybe they should join TIL?

**Review by miniksa — APPROVED** [review]
Works for me. Awesome cleanup. Much EIM.

File: src/cascadia/TerminalApp/TabRowControl.h (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TabRowControl.h:27]: > so we understand which ones are a part of our magic and which ones are a part of the actual cppwinrt packages.

> implying anything useful comes with cppwinrt

**Pull request merged**